src/worker/mod.rs

```diff
@@ -263,6 +275,34 @@ impl Worker {
         task_token: TaskToken,
         status: activity_result::Status,
     ) -> Result<(), CompleteActivityError> {
+        if task_token.is_local_activity_task() {
+            if let Some(la_info) = self.local_act_mgr.complete(&task_token).await {
+                if let Err(e) = self
+                    .wft_manager
+                    .notify_of_local_result(
+                        &la_info.workflow_execution.run_id,
+                        la_info.seq,
+                        LocalResolution::LocalActivity(ActivityResult {
+                            status: Some(status),
+                        }),
+                    )
+                    .await
+                {
+                    error!(
+                        "Problem completing local activity {}: {:?} -- will evict the workflow",
+                        task_token, e
+                    );
+                    self.request_wf_eviction(
```

Comment:
This doesn't really look like it belongs here.
The workflow may have already been evicted for other reasons at this point no?
What kind of errors could `notify_of_local_result` return?

Reply:
The evict call is idempotent so if it's already been requested that's fine. `notify_of_local_result` can return any of the state machine errors since it's literally updating the LA machine state.